### DISCUSSION TITLE
upgrade from old 1.28.0-dev (adriankumpf/teslamate) to latest, how? help.

### DISCUSSION THREAD

**Discussion by Zorkny** [teslamate-org/teslamate#4100] [Q&A]
Hi,

I have been running 1.28.0-dev from adriankump/teslamate for a long time and are now trying to upgrade.
Its a big step to latest and it seems to fail in many ways.

Before I go into each error, any pointers or hints on what to do?

Thanks!

**Comment by cwanja** [comment]
How does it fail? It should upgraded. 

Although if I remember, some people had to update their underlying OS depending on what you were running. Take that into consideration.

**Comment by Zorkny** [comment]
OS is updated with latest, but I think its related with the old containers being named root-teslamate-1, root-mosquitto-1, root-grafana-1, root-database-1. The new are named "teslamate-..." when I upgrade and there are conflicts.
I run docker commands as root.

**Reply by cwanja** [reply]
Can you take a backup and wipe out the current containers? Then restore the backup.

**Reply by cwanja** [reply]
Advise against running as root.

**Reply by Zorkny** [reply]
But wouldnt the docker-compose file make it run as another user?

**Reply by Zorkny** [reply]
Other docker files have "user: 1000:1000 # default" or similar,  but not the teslamate ones?

**Reply by cwanja** [reply]
That depends on how you setup your OS and runtime. I highly suggest not using root to run apps and creating files unless absolutely necessary. 

Create yourself a user and let that user create the files and containers.

**Reply by Zorkny** [reply]
Thanks, corrected this and migrated all to a different user. It was the volume names that were different too, hade to rename them before the upgrade. Will let it run for some days to see that it works. Thanks for quick support!

**Comment by Zorkny** [comment]
Solved by migrating to non-root user and moving volumes to correct names before upgrading. Now upgraded to latest version ok.
